core/kazoo_documents/src/kzd_blacklists.erl

```diff
@@ -5,23 +5,271 @@
+        {'ok', []} -> {'error', 'not_found'};
+        {'ok', JObjs} -> format_view_results(JObjs, Options);
+        {'error', _}=_E ->
+            lager:error("error getting blacklist patterns for account ~s and owner ~s : ~p", [AccountId, OwnerId, _E]),
+            {'error', 'not_found'}
+    end.
+
+-spec fetch_number(kz_term:ne_binary(), kz_term:api_ne_binary(), kz_term:ne_binary()) -> {'ok', kz_json:objects()} | {'error', 'not_found'}.
+fetch_number(AccountId, OwnerId, Number) ->
+    fetch_number(AccountId, OwnerId, Number, #{}).
+
+-spec fetch_number(kz_term:ne_binary(), kz_term:api_ne_binary(), kz_term:ne_binary(), map()) -> {'ok', kz_json:objects()} | {'error', 'not_found'}.
+fetch_number(AccountId, OwnerId, Number, Options) ->
+    Db = kz_util:format_account_db(AccountId),
+    ViewOptions = get_view_options(?LIST_BY_OWNER, OwnerId, <<"number">>, Number),
+    case kz_datamgr:get_results(Db, ?LIST_BY_OWNER, ViewOptions) of
+        {'ok', []} -> {'error', 'not_found'};
+        {'ok', [JObj]} ->
+            JObjs = kz_json:get_list_value(<<"value">>, JObj),
+            case filter_results(JObjs, Options) of
+                [] -> {'error', 'not_found'};
+                Results -> {'ok', Results}
+            end;
+        {'error', _}=_E ->
+            lager:error("error getting blacklists by number ~s for account ~s and owner ~s : ~p", [Number, AccountId, OwnerId, _E]),
+            {'error', 'not_found'}
+    end.
+
+-spec filter_results(kz_json:objects(), map()) -> kz_json:objects().
+filter_results(JObjs, Options) ->
+    Routines = get_view_filters(Options),
+    lists:foldr(fun(F, Item) -> F(Item) end
+               ,JObjs
+               ,Routines
+               ).
+
+-spec get_view_options(kz_term:ne_binary(), kz_term:api_ne_binary(), kz_term:ne_binary()) -> kz_term:proplist().
+get_view_options(?LIST_BY_OWNER, 'undefined', <<"pattern">> = Type) ->
+    props:filter_undefined(
+      [{'group', 'true'}
+      ,{'group_level', 3}
+      ,{'startkey', ['null', Type]}
+      ,{'endkey', ['null', Type, kz_json:new()]}
+      ]);
+get_view_options(?LIST_BY_OWNER, OwnerId, <<"pattern">> = Type) ->
+    props:filter_undefined(
+      [{'group', 'true'}
+      ,{'group_level', 3}
+      ,{'startkey', [OwnerId, Type]}
+      ,{'endkey', [OwnerId, Type, kz_json:new()]}
+      ]).
+
+-spec get_view_options(kz_term:ne_binary(), kz_term:api_ne_binary(), kz_term:api_ne_binary(), kz_term:api_ne_binary()) -> kz_term:proplist().
+get_view_options(?LIST_BY_OWNER, 'undefined', <<"number">> = Type, Number) ->
+    props:filter_undefined(
+      [{'group', 'true'}
+      ,{'group_level', 3}
+      ,{'key', ['null', Type, Number]}
+      ]);
+get_view_options(?LIST_BY_OWNER, OwnerId, <<"number">> = Type, Number) ->
+    props:filter_undefined(
+      [{'group', 'true'}
+      ,{'group_level', 3}
+      ,{'key', [OwnerId, Type, Number]}
+      ]).
+
+-spec format_view_results(kz_json:objects(), map()) -> kz_json:object().
```

Comment:
return type is wrong, this does not return `kz_json:object()` but `{'ok', kz_json:object()} | {'error', 'not_found'}`